api/types.go

```diff
@@ -691,6 +718,7 @@ func (pin *Pin) ProtoMarshal() ([]byte, error) {
 		// UserAllocations:      pin.UserAllocations,
 		Metadata:  pin.Metadata,
 		PinUpdate: pin.PinUpdate.Bytes(),
+		ExpireAt:  expireAtProto,
```

Comment:
Given that the protobuf timestamp type is a pointer, leave it as null when the expireAt is less than now. This will very probably save lots of space by not having to store an actual time.